### PR TITLE
Remove some warnings due to generics

### PR DESCRIPTION
**Pull request by stephanenicolas**
Following up #30, this PR fixes a few warnings in code due to generics or varargs.

### CODE REVIEW

File: grox-sample-rx/src/main/java/com/groupon/grox/sample/MainActivity.java [hunk 1]
```diff
@@ -39,13 +40,17 @@ protected void onCreate(Bundle savedInstanceState) {
...
 
     subscription.add(
         clicks(button)
             .map(click -> new RefreshColorCommand())
             .flatMap(Command::actions)
-            .subscribe(store::dispatch));
+            .subscribe(store::dispatch, this::doLog));
+  }
+
+  private void doLog(Throwable throwable) {
```
Comment (runningcode) [inline grox-sample-rx/src/main/java/com/groupon/grox/sample/MainActivity.java]: this is a small behavior change. before the app would crash, now it is just logged to logcat.
IMO, i think its better to crash the app because i don't always check logcat.

File: grox-core/src/main/java/com/groupon/grox/Store.java [hunk 2]
```diff
@@ -53,10 +52,21 @@
...
-  public Store(STATE initialState, Middleware<STATE>... middlewares) {
+  public Store(STATE initialState) {
     this.state = initialState;
     this.middlewares.add(new NotifySubscribersMiddleware());
-    this.middlewares.addAll(asList(middlewares));
+    this.middlewares.add(new CallReducerMiddleware());
+  }
+
+  @SafeVarargs
+  public Store(STATE initialState, Middleware<STATE> first, Middleware<STATE>... others) {
```
Comment (alin-turcu) [inline grox-core/src/main/java/com/groupon/grox/Store.java]: why this change?

File: grox-sample-rx/src/main/java/com/groupon/grox/sample/MainActivity.java (hunk 1 above)
Comment (stephanenicolas) [reply grox-sample-rx/src/main/java/com/groupon/grox/sample/MainActivity.java]: @runningcode , I just added this for the sample. We have a RxLint check and it was red. On our side, in prod, we log it back to our servers.

File: grox-core/src/main/java/com/groupon/grox/Store.java (hunk 2 above)
Comment (stephanenicolas) [reply grox-core/src/main/java/com/groupon/grox/Store.java]: There are 2 constructors, 1 with no middlewares, and the other one with middle wares and in this case, one is mandatory now. This litte trick helps the compiler to disambiguate the calls better.

But let me check something before merging it..
